PROJECT

```diff
@@ -1,16 +1,44 @@
+domain: metal3.io
+layout:
+- go.kubebuilder.io/v3
 multigroup: true
+projectName: baremetal-operator
 repo: github.com/metal3-io/baremetal-operator
 resources:
-- group: metal3.io
+- api:
+    crdVersion: v1
+    namespaced: true
+  controller: true
+  domain: metal3.io
+  group: metal3.io
   kind: BareMetalHost
+  path: github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1
```

Comment:
I have generated the webhook file initially using kubebuilder as instructed [here](https://book.kubebuilder.io/reference/webhook-overview.html?highlight=webhook#supporting-older-cluster-versions) which didn't add webhook in the Project file automatically. Not sure if we should add it manually here.

Reply:
yes Kubebuilder by default doesn't do it and generates very minimal  file. So, since we have the webhook enabled, I would suggest that we add it
```
  webhooks:
    validation: true
    webhookVersion: v1
```

Reply:
Not sure if thats a necessity. Some examples:
1. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/PROJECT -> V3 project
2. https://github.com/kubernetes-sigs/cluster-api/blob/main/PROJECT -> v2 project
3. https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/PROJECT - > v2 project

My point on not touching it manually being, if we are modifying this file manually, I see no point of this PR itself since we could have manually modified other files as well as we were doing.

Reply:
ok sure

Reply:
@fmuyassarov I have tried adding the webhook generation through kubebuilder from scratch again and I found out I didnt use the right webhook flags while I generated it. When I added the right flags, I can see the changes added in the PROJECT file automatically. Thanks for pointing it out. I have added a separate commit for this. PTAL again.